Content.Server/Explosion/TriggerSystem.cs

```diff
@@ -72,12 +84,16 @@ public void Explode(EntityUid uid, ExplosiveComponent component)
 
         #region Flash
         private void HandleFlashTrigger(EntityUid uid, FlashOnTriggerComponent component, TriggerEvent args)
-                {
-                    if (component.Flashed) return;
+        {
+            bool repeatable = component.Repeating && component.LastFlash + TimeSpan.FromSeconds(component.Cooldown) < _gameTiming.CurTime;
 
-                    FlashableComponent.FlashAreaHelper(component.Owner, component.Range, component.Duration);
-                    component.Flashed = true;
-                }
+            if (!component.Flashed ^ repeatable)
+            {
+                FlashableComponent.FlashAreaHelper(component.Owner, component.Range, component.Duration);
+                component.Flashed = true;
```

Comment:
isnt flashed completely useless since you added lastflash? might as well axe it imo

Reply:
disregard, do what sloth said